test/FIFOMarket.test.ts

```diff
@@ -614,6 +614,38 @@ describe('FIFOMarket', () => {
           .toString()
       );
     });
+    it('should sell removals in order of vintage regardless of minting order', async () => {
+      const buyerInitialBPNoriBalance = formatTokenAmount(1_000_000);
+      const { bpNori, fifoMarket, hre, totalAmountOfRemovals } =
+        await setupTestLocal({
+          buyerInitialBPNoriBalance,
+          removalDataToList: [
+            { amount: 1, vintage: 2015 },
+            { amount: 5, vintage: 2014 },
+          ],
+        });
+      const { buyer } = hre.namedAccounts;
+
+      const purchaseAmount = '4';
+      const fee = '.15';
+      const totalPrice = (Number(purchaseAmount) + Number(fee)).toString();
+
+      await bpNori
+        .connect(hre.namedSigners.buyer)
+        .send(
+          fifoMarket.address,
+          hre.ethers.utils.parseUnits(totalPrice),
+          hre.ethers.utils.hexZeroPad(buyer, 32)
+        );
+
+      const totalNumberActiveRemovals =
+        await fifoMarket.totalNumberActiveRemovals();
+
+      // Roundabout way of showing that the removal used to fill the order was the removal with amount 5, not 1.
+      expect(totalNumberActiveRemovals.toNumber()).to.equal(
+        totalAmountOfRemovals
+      );
+    });
```

Comment:
Test case idea: mint removals that belong to a few vintages, sell through all of a vintage's removals and into the next vintage.  see that nothing crashes. maybe try to make another purchase and see that it picks up from the next expected earliest year. (as you mention, the first case is covered by our existing tests, maybe check that the second case of making another purchase AND checking that the removals used to fulfill it are as expected is being covered)